Validation/HGCalValidation/test/HGCMissingRecHit.cc

```diff
@@ -0,0 +1,458 @@
+    analyzeHGCalDigi(theHits, 2, bhHitRefs);
+  } else {
+    edm::LogWarning("HGCalValid") << "No BH Digi Found " << std::endl;
+  }
+
+  //accessing EE Rechit information
+  const edm::Handle<HGCeeRecHitCollection> &eeRecHit = iEvent.getHandle(eeRecHitToken_);
+  if (eeRecHit.isValid()) {
+    const HGCeeRecHitCollection *theHits = (eeRecHit.product());
+    analyzeHGCalRecHit(theHits, 0, eeHitRefs);
+  } else {
+    edm::LogWarning("HGCalValid") << "No EE RecHit Found " << std::endl;
+  }
+
+  //accessing FH Rechit information
+  const edm::Handle<HGChefRecHitCollection> &fhRecHit = iEvent.getHandle(fhRecHitToken_);
+  if (fhRecHit.isValid()) {
+    const HGChefRecHitCollection *theHits = (fhRecHit.product());
+    analyzeHGCalRecHit(theHits, 1, fhHitRefs);
+  } else {
+    edm::LogWarning("HGCalValid") << "No FH RecHit Found " << std::endl;
+  }
+
+  //accessing BH Rechit information
+  const edm::Handle<HGChebRecHitCollection> &bhRecHit = iEvent.getHandle(bhRecHitToken_);
+  if (bhRecHit.isValid()) {
+    const HGChebRecHitCollection *theHits = (bhRecHit.product());
+    analyzeHGCalRecHit(theHits, 2, bhHitRefs);
+  } else {
+    edm::LogWarning("HGCalValid") << "No BH RecHit Found " << std::endl;
+  }
+}
+
+void HGCMissingRecHit::analyzeHGCalSimHit(edm::Handle<std::vector<PCaloHit>> const &simHits,
+                                          int idet,
+                                          std::map<unsigned int, HGCHitTuple> &hitRefs) {
+  for (auto const &simHit : *simHits) {
+    DetId id(simHit.id());
+    bool ok(true);
+    GlobalPoint p;
+    std::ostringstream st1;
+    if (id.det() == DetId::HGCalHSc) {
+      st1 << HGCScintillatorDetId(id);
+    } else if ((id.det() == DetId::HGCalEE) || (id.det() == DetId::HGCalHSi)) {
+      st1 << HGCSiliconDetId(id);
+    } else {
+      st1 << "Not a Standard One";
+    }
+    if ((hgcCons_[idet]->waferHexagon8()) && ((id.det() == DetId::HGCalEE) || (id.det() == DetId::HGCalHSi))) {
+      p = hgcGeometry_[idet]->getPosition(id);
+    } else if ((hgcCons_[idet]->tileTrapezoid()) && (id.det() == DetId::HGCalHSc)) {
+      p = hgcGeometry_[idet]->getPosition(id);
+      edm::LogVerbatim("HGCalGeom") << "Scint " << HGCScintillatorDetId(id) << " position (" << p.x() << ", " << p.y()
+                                    << ", " << p.z() << ")";
+    } else {
+      // This is an invalid cell
+      ok = false;
+      edm::LogVerbatim("HGCalError") << "Hit " << std::hex << id.rawId() << std::dec << " " << st1.str()
+                                     << " in the wrong collection for detector " << idet << ":" << geometrySource_[idet]
+                                     << " ***** ERROR *****";
+    }
+
+    edm::LogVerbatim("HGCalValid") << "SimHit: " << std::hex << id.rawId() << std::dec << " " << st1.str() << " Flag "
+                                   << ok;
+
+    if (ok) {
+      float xp = p.x();
+      float yp = p.y();
+      float zp = p.z();
+      float energy = simHit.energy();
```

Comment:
Would it make sense to store directly a `GlobalPoint`? That would avoid implementing yet another `getEta` method a few lines below since a `GlobalPoint` has an `eta()` method.